lib/opam_cmd.ml

```diff
@@ -212,16 +212,17 @@ let get_opam_info ~root ~pins packages =
 let package_is_valid { package; dev_repo; _ } =
   match dev_repo with
   | `Error msg ->
-      R.error_msg (Fmt.strf "Do not know how to deal with %a: %s" pp_package package msg)
+      Logs.warn (fun l -> l "Do not know how to deal with %a: %s" pp_package package msg);
```

Comment:
How about moving the logs in a different function so that we only print logs about such packages if we explicit want to instead of it being a site effect of computing this `is_valid` property?

I.e. we would have a:
```ocaml
let log_invalid_packages packages =
  List.iter
   (function
      | {dev_repo = `Error msg; _} -> ...
      | {dev_repo = `Unknown host; _ } -> ...
      | _ ->  ())
   packages
```
That we would call at the appropriate time during the command execution.

Reply:
Done